layers/cmd/ingest_csv.go

```diff
@@ -30,6 +30,98 @@ import (
 	"github.com/cloudprivacylabs/opencypher/graph"
 )
 
+type CSVIngester struct {
+	BaseIngestParams
+	StartRow     int
+	EndRow       int
+	HeaderRow    int
+	ID           string
+	InitialGraph string
+}
+
+func (ci CSVIngester) Run(pipeline *PipelineContext, file string) {
+	// ctx := ls.NewContext(context.WithValue(context.Background(), "filename", file))
+	ctx := ls.DefaultContext()
+	// pipeline.Input = strings.NewReader(context.Background().Value("filename").(string))
```

Comment:
Don't create new context. Embed ls.Context into PipelineContext, and use the pipeline context instead of ctx.